test/helpers.py

```diff
@@ -22,8 +22,8 @@ def to_char_p_p(options: List[str]):
   c_options[:] = [ctypes.cast(ctypes.create_string_buffer(o.encode("utf-8")), ctypes.POINTER(ctypes.c_char)) for o in options]
   return c_options
 
-def cuda_compile(prg, options, f, check):
-  check(f.create(ctypes.pointer(prog := f.new()), prg.encode(), "<null>".encode(), 0, None, None))
+def compile(prg, options, f, check, filename="<null>".encode()):
+  check(f.create(ctypes.pointer(prog := f.new()), prg.encode(), filename, 0, None, None))
```

Comment:
I'm fine with the filename parameter, but maybe put the `.encode()` in the function?